lib/FormCustomProperties/CustomPropertiesListField.js

```diff
@@ -216,18 +198,51 @@ export default class CustomPropertiesListField extends React.Component {
       });
     };
 
-    return (
-      <FieldComponent
-        data-test-customproperty-value
-        error={!isEmpty(errors) ? errors[customProperty.value] : undefined}
-        id={`edit-customproperty-${i}-value`}
-        label={<FormattedMessage id="stripes-erm-components.prop.customPropertyValue" />}
-        onChange={handleChange}
-        value={controlledFieldValue}
-        {...fieldProps}
-        required={!customProperty.primary}
-      />
-    );
+    return (customProperty.type === customPropertyTypes.NUMBER || customProperty.type === customPropertyTypes.DECIMAL) ?
```

Comment:
This can be done a lot cleaner a few different ways that we already make use of. Eg, grouping shared props and spreading them like with `fieldProps`.

Given the size that this function is getting to, it might also be worth breaking it out into its own component. Then we wouldn't need to do an IntlConsumer wrap on one part of a ternary expression either (though I really would like to avoid this ternary in general).

Reply:
Yeah, was going to clean it up, but yeah will move into a different component so that `intl` can be passed down